File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/impl/nms/listener/packets/serverbound/NmsServerboundPacketImpl.java
package dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.serverbound;

import dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.NmsPacketImpl;
import net.minecraft.network.protocol.Packet;
import net.minecraft.network.protocol.game.ServerGamePacketListener;


public abstract class NmsServerboundPacketImpl<Nms extends Packet<ServerGamePacketListener>> extends
    NmsPacketImpl<Nms, ServerGamePacketListener> {

  protected NmsServerboundPacketImpl(Nms nmsPacket) {
    super(nmsPacket);
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/impl/nms/listener/packets/serverbound/NmsServerboundPacketImpl.kt
package dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.serverbound

import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.NmsPacketImpl
import net.minecraft.network.protocol.Packet
import net.minecraft.network.protocol.common.ServerCommonPacketListener

@NmsUseWithCaution
abstract class NmsServerboundPacketImpl<Nms : Packet<out ServerCommonPacketListener>>(nmsPacket: Nms) :
    NmsPacketImpl<Nms, ServerCommonPacketListener>(nmsPacket)
